docs/reference/secrets.md

```diff
@@ -0,0 +1,65 @@
+# Using secrets
+
+This page shows how to use secrets within your functions for API tokens, passwords and similar.
+
+Using secrets is a two step process. First we need to define the secret in your cluster and then you need to 'use' the secret to your function. You can find a simple example function [ApiKeyProtected in the OpenFaaS repo](https://github.com/openfaas/faas/tree/master/sample-functions/ApiKeyProtected-Secrets). When we deploy this function we provide a secret key that it uses to authenticate requests.
+
+_Note_: The examples in the following section require `faas-cli` version `>=0.5.1`
```

Comment:
Good note. Was this added historically? Is it now safe to assume people are already on this version?

Reply:
I pulled a lot of this from the secrets walkthrough in the faas repo, we added it awhile back.  I honestly don't know if it is needed, I would be surprised if people were using these docs and not already on a compatible version.  I figured it was safest to start including it and then decide if we should remove it